File: apis/src/main/java/com/beat/apis/performance/application/dto/makerPerformance/MakerPerformanceDetailResponse.java
package com.beat.apis.performance.application.dto.makerPerformance;

import com.beat.global.support.jackson.CdnImageUrl;

public record MakerPerformanceDetailResponse(
	Long performanceId,
	String genre,
	String performanceTitle,
	@CdnImageUrl String posterImage,
	String performancePeriod,
	int minDueDate
) {
	public static MakerPerformanceDetailResponse of(
		Long performanceId,
		String genre,
		String performanceTitle,
		String posterImage,
		String performancePeriod,
		int minDueDate) {  // minDueDate 매개변수 추가
		return new MakerPerformanceDetailResponse(performanceId, genre, performanceTitle, posterImage,
			performancePeriod, minDueDate);
	}
}

File: apis/src/main/kotlin/com/beat/apis/performance/api/response/MakerPerformanceDetailResponse.kt
package com.beat.apis.performance.api.response

import com.beat.application.frontoffice.performance.maker.query.MakerPerformanceResult
import com.beat.apis.web.jackson.CdnImageUrl

@ConsistentCopyVisibility
data class MakerPerformanceDetailResponse private constructor(
    val performanceId: Long?,
    val genre: String?,
    val performanceTitle: String?,
    @field:CdnImageUrl val posterImage: String?,
    val performancePeriod: String?,
    val minDueDate: Int,
) {
    companion object {
        fun from(result: MakerPerformanceResult): MakerPerformanceDetailResponse = MakerPerformanceDetailResponse(
            performanceId = result.performanceId,
            genre = result.genre,
            performanceTitle = result.performanceTitle,
            posterImage = result.posterImage,
            performancePeriod = result.performancePeriod,
            minDueDate = result.minDueDate,
        )
    }
}
